### PR TITLE
implement tuple let pattern

### PR DESCRIPTION
**Pull request by StachuDotNet**
Changelog:

```
Language and Standard Library
- add tuple deconstruction in `let` bindings (i.e. `let (a, b) = (1, "two")` works)
```

Extracted from #4711

### CODE REVIEW

**Comment by StachuDotNet** [comment]
Ah, todo: add serialization roundtripping and consistency (persisted) tests.

File: backend/testfiles/execution/language.tests [hunk 1]
```diff
@@ -13,6 +13,11 @@ module Let =
   (let x = 5 in x) = 5
   (let x = 5 in let x = 6 in x) = 6
 
+  (let (a, b) = (1, 2) in 2) = 2
```
Comment (pbiggar) [inline backend/testfiles/execution/language.tests:16]: Worth checking length of 3 and 4, and also errors

File: backend/src/LibExecution/Interpreter.fs [hunk 2]
```diff
@@ -86,22 +86,75 @@ let rec eval' (state : ExecutionState) (st : Symtable) (e : Expr) : DvalTask =
     | ECharacter (_id, s) -> return DChar s
 
     | ELet (id, pattern, rhs, body) ->
-      match pattern with
-      | LPVariable (_id, lhs) ->
-        if lhs = "" then
-          let! _ = preview st rhs
-          return DError(sourceID id, "Variable name in `let` is empty")
+      /// Returns `incomplete` traces for subpatterns of an unmatched pattern
```
Comment (pbiggar) [inline backend/src/LibExecution/Interpreter.fs:89]: Hard to tell if this works from a read of it, but there's a test case for matchPatterns that could be copied to ensure the analysis works.

**Comment by nicu-chiciuc** [comment]
I managed to rebase the current branch: https://github.com/darklang/dark/compare/main...nicu-chiciuc:dark:tuple-let-pattern-rebase

I tried to maintain the logic of the commits as it was before

It builds ok, but I don't yet know how to regenerate `backend/serialization/oplist-binary-latest.bin` or `backend/serialization/oplist-binary-pretty-c8ae15a75b156eae0782f6149e7004afcdf3e823.json`.

**Comment by StachuDotNet** [comment]
> It builds ok, but I don't yet know how to regenerate...

`./scripts/run-backend-tests` will refresh those files for you. (caused by backend/tests/Tests/Serialization.Vanilla.fs and Serialization.Binary.fs

**Comment by StachuDotNet** [comment]
ah, you've already figured that out! :)

**Comment by StachuDotNet** [comment]
#4840 supersedes this, closing. Thanks

**Pull request closed without merging**